### PR TITLE
Test member data

### PR DESCRIPTION
**Pull request by ondenman**
Tests scraper output against YAML file.
MemberPage#to_h should match the data specified in the YAML.

Part of everypolitician/everypolitician#600 (Ensure every scraper that has been ported to Scraped has a basic YAML test for at least one person)

This PR replaces https://github.com/everypolitician-scrapers/nigeria-national-assembly/pull/17

### CODE REVIEW

**Review by tmtmtmtm — CHANGES_REQUESTED** [review]
@ondenman can I ask you to give this a better title? At the minute it's a bit of a word salad, which makes it quite hard to know what it's meant to be doing. Is this testing the "member data refactor"? Refactoring the "test member data"? Neither of them seem to match what's actually going on…

File: lib/member_page.rb [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    party_node_match.captures.first
+  end
+
+  field :party_id do
+    party_node_match.captures.last
+  end
+
+  field :image do
+    image_url = box.css('.carousel-inner img/@src').text
+    image_url.include?('/avatar.jpg') ? nil : image_url
```
Comment (tmtmtmtm) [inline lib/member_page.rb:55]: Would be good to make sure this is covered

File: lib/member_page.rb [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  end
+
+  field :source do
+    url.to_s
+  end
+
+  field :area do
+    constituency.empty? ? '' : [constituency, state].join(', ')
```
Comment (tmtmtmtm) [inline lib/member_page.rb:63]: the 'empty' side of this doesn't seem to be covered

File: lib/member_page.rb [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+  def party_node
+    box.xpath('.//th[text()="Party"]/following-sibling::td').text
+  end
+
+  def party_node_match
+    party_node.match(/^(.*)\s+\((.*)\)\s*$/) || abort("Bad party: #{party_node}")
+  end
+
+  TITLE_RE = /^(Hon\.|Sen\.) /
```
Comment (tmtmtmtm) [inline lib/member_page.rb:84]: Both tests at the minute are for Hon., and none for Sen.

**Review by tmtmtmtm — CHANGES_REQUESTED** [review]
Generally looking good, but I think we need a few more test cases. In general we should make sure that any combination we explicitly code for is covered somewhere, to avoid any regressions later on those parts.

File: lib/member_page.rb (hunk 3 above)
Comment (ondenman) [reply lib/member_page.rb:84]: I've added the additional tests and will rebase to bring the branch up to date with master.

File: lib/member_page.rb (hunk 1 above)
Comment (ondenman) [reply lib/member_page.rb:55]: Added in aacd30d

File: lib/member_page.rb (hunk 2 above)
Comment (ondenman) [reply lib/member_page.rb:63]: Added in 6763788

File: lib/member_page.rb (hunk 3 above)
Comment (ondenman) [reply lib/member_page.rb:84]: Added in 77bebab

**Review by tmtmtmtm — APPROVED** [review]

**Pull request merged**